airflow/providers/papermill/operators/papermill.py

```diff
@@ -45,19 +45,26 @@ class PapermillOperator(BaseOperator):
     :param parameters: the notebook parameters to set
     :type parameters: dict
     """
+    supports_lineage = True
+
     @apply_defaults
     def __init__(self,
-                 input_nb: str,
-                 output_nb: str,
-                 parameters: Dict,
+                 input_nb: Optional[str] = None,
+                 output_nb: Optional[str] = None,
+                 parameters: Optional[Dict] = None,
                  *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
-        self.inlets.append(NoteBook(url=input_nb,
-                                    parameters=parameters))
-        self.outlets.append(NoteBook(url=output_nb))
+        if input_nb:
+            self.inlets.append(NoteBook(url=input_nb,
+                                        parameters=parameters))
+        if output_nb:
+            self.outlets.append(NoteBook(url=output_nb))
 
     def execute(self, context):
+        if not self.inlets or not self.outlets:
+            raise ValueError("Input notebook or output notebook is not specified")
```

Comment:
I don't like it. We should raise this error in __init__ to inform users as soon as possible that their DAG will fail if they do not specify inlets and outlets.

Reply:
I agree but can you propose how to do that if with short hand notation inlets/outlets are set after initialization has taken place of the operator?

Reply:
I think it's ok to have it at execute time - no matter if we use builder pattern or the shorthand operators, it cannot be done init().